machine.c

```diff
@@ -342,7 +350,17 @@ void telestratwrite( struct m6502 *cpu, unsigned short addr, unsigned char data
   {
     switch( addr & 0x0f0 )
     {
-      case 0x20:
+
+      case 0x40:
+        if (oric->ch376_activated)
+        {
+          if (addr == 0x340 || addr == 0x341)
+            ch376_oric_write(oric->ch376, addr, data);
+        }
+      break;
```

Comment:
This isn't quite correct. If ch476 is NOT activated, writes to 0x34x do nothing, instead of the normal VIA write.

Please move the 0x40 case to just before the "default:" and only do "break" inside the if(), that way if ch376 is NOT enabled, it'll drop into the default case.